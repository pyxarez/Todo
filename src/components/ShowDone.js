import React from 'react';

import './ShowDone.res/style.css';

ShowDone.propTypes = {
  handleShowDoneClick: React.PropTypes.func.isRequired,
  showDone: React.PropTypes.bool
}

export default function ShowDone({handleShowDoneClick, showDone}) {
  return (
    <label className="my-showdone-component">
      <input onChange={handleShowDoneClick} checked={showDone} type="checkbox"/>
      Show done
    </label>
    );
}