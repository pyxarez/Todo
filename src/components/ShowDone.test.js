import React from 'react';
import { shallow } from 'enzyme';

import ShowDone from './ShowDone';

const setup = () => {
  const props = {
    handleShowDoneClick: () => {},
    ShowDone: true
  }
  const wrapper = shallow(<ShowDone {...props}/>)

  return {
    props,
    wrapper
  }
}

describe('Components :: ShowDone', () => {
  it('should render self', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});