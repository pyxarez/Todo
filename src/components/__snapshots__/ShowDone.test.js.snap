exports[`Components :: ShowDone should render self 1`] = `
<label
  className="my-showdone-component">
  <input
    onChange={[Function]}
    type="checkbox" />
  Show done
</label>
`;
